internal/pod/pod.go

```diff
@@ -0,0 +1,141 @@
+package pod
+
+import (
+	"context"
+	"fmt"
+
+	kmmv1beta1 "github.com/kubernetes-sigs/kernel-module-management/api/v1beta1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+//go:generate mockgen -source=pod.go -package=pod -destination=mock_pod.go
+
+type ImagePuller interface {
+	CreatePullPod(ctx context.Context, imageSpec *kmmv1beta1.ModuleImageSpec, micObj *kmmv1beta1.ModuleImagesConfig) error
+	DeletePod(ctx context.Context, pod *v1.Pod) error
+	ListPullPods(ctx context.Context, micObj *kmmv1beta1.ModuleImagesConfig) ([]v1.Pod, error)
+	GetPullPodForImage(pods []v1.Pod, image string) *v1.Pod
+}
+
+const (
+	moduleImageLabelKey = "kmm.node.kubernetes.io/module-image-config"
+	ImageLabelKey       = "kmm.node.kubernetes.io/module-image"
```

Comment:
WDYT if we make the ImageLabelKey also private, and add an interface function that returns the image of the pullPod?

Reply:
Done.